### PR TITLE
Refactor process documentation

### PR DESCRIPTION
**Pull request by lbernick**
This commit makes several changes:
- splits contributor ladder into a separate section, since it's very long and makes the process doc harder to read
- moves the TEP process docs in with the other process docs. This also makes the TEP process doc easier to update, since you don't have to update both the markdown and the template file.
- moves the docs on Tekton projects into a separate folder, since we plan to add new guidance on adopting existing projects into the Tekton org

This commit doesn't propose any process changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,14 +29,14 @@ See our standards regarding:
...
 
-- [Find something to work on](process.md#finding-something-to-work-on)
-- [Propose features](process.md#proposing-features)
-- [Contributor ladder](process.md#contributor-ladder)
-- [Pull request reviews](process.md#reviews)
-- [Propose projects](process.md#proposing-projects)
+- [Find something to work on](./process/README.md#finding-something-to-work-on)
+- [Propose features](./process/README.md#proposing-features)
+- [Contributor ladder](./process/contributor-ladder.md)
+- [Pull request reviews](./process/README.md#reviews)
```
Comment (jerop) [inline README.md:35]: I suggest also linking pull request process? like you did in process.README.md?

**Review by jerop — APPROVED** [review]
thanks @lbernick!

File: README.md (hunk 1 above)
Comment (lbernick) [reply README.md]: done, thanks!

File: teps/README.md [hunk 2]
```diff
@@ -1,185 +1,8 @@
-# Tekton Enhancement Proposals (TEPs)
+# TEPs
```
Comment (afrittoli) [inline teps/README.md:1]: NIT: I think the title here should stay as it was

**Review by afrittoli — COMMENTED** [review]
Thank you @lbernick - this looks really good to me.
Just a nit on the title, but we can also revert that as a follow up.

File: teps/README.md (hunk 2 above)
Comment (lbernick) [reply teps/README.md]: fixed, thanks!

**Comment by jerop** [comment]
@dibyom seems I can't approve, want to trade places? 😂

**Pull request merged**